libraries/zookeeper_config.rb

```diff
@@ -67,9 +65,22 @@ def to_s
             mode '0644'
           end
 
-          file new_resource.path do
+          # create zoo.properties.static file and only update 
+          # zoo.properties when static file has changed.
+          # this is needed to ensure chef won't constantly update
+          # zoo.properties and force restarts of zookeeper when
+          # it dynamically updates zoo.properties.
+          file new_resource.path+".static" do
```

Comment:
Why is the file continually updating? This mechanism won't change that it just makes it more obfuscated.